src/Goetas/Twital/EventSubscriber/IDNodeSubscriber.php

```diff
@@ -0,0 +1,52 @@
+<?php
+namespace Goetas\Twital\EventSubscriber;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Goetas\Twital\EventDispatcher\TemplateEvent;
+use Goetas\Twital\Twital;
+
+/**
+ *
+ * @author Asmir Mustafic <goetas@gmail.com>
+ *
+ */
+class IDNodeSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return array(
+            'compiler.post_load' => array(
+                array(
+                    'addAttribute'
+                )
+            ),
+            'compiler.pre_dump' => array(
+                array(
+                    'removeAttribute'
+                )
+            )
+        );
+    }
+
+    public function addAttribute(TemplateEvent $event)
+    {
+        $doc = $event->getTemplate()->getDocument();
+        $xp = new \DOMXPath($doc);
+        $nodes = $xp->query("//*[@*[namespace-uri()='".Twital::NS."']]");
+        $cnt = 0;
+        foreach ($nodes as $node) {
+            $node->setAttributeNS(Twital::NS, '__internal-id__', ++$cnt);
```

Comment:
> What about using here uniqid ?

Simple `rand`might be better in this case. If uniqid is called fast enough, it will return the same id. There could also be md5 from uniqid, rand and spl_object_hash, but it would probably be overkill :)